foundation/foundation-configuration-store/src/main/java/com/boozallen/aissemble/configuration/policy/PropertyRegenerationPolicyManager.java

```diff
@@ -0,0 +1,190 @@
+import com.boozallen.aissemble.configuration.policy.exception.PropertyRegenerationPolicyException;
+import com.boozallen.aissemble.configuration.store.PropertyKey;
+import com.boozallen.aissemble.core.policy.configuration.policy.DefaultPolicy;
+import com.boozallen.aissemble.core.policy.configuration.policy.Policy;
+import com.boozallen.aissemble.core.policy.configuration.policy.Target;
+import com.boozallen.aissemble.core.policy.configuration.policy.json.PolicyInput;
+import com.boozallen.aissemble.core.policy.configuration.policy.json.rule.PolicyRuleInput;
+import com.boozallen.aissemble.core.policy.configuration.policymanager.AbstractPolicyManager;
+
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import org.apache.commons.lang3.StringUtils;
+
+/**
+ * Class used for loading and configuring policies for determining when to regenerate a given {@link Property}.
+ */
+public class PropertyRegenerationPolicyManager extends AbstractPolicyManager {
+
+    private static final String TARGETS_REGEX = "/([^/]+)/([^/]+)";
+
+    private Set<PropertyRegenerationPolicy> propertyRegenerationPolicies;
+
+    /**
+     * Gets the property regeneration specific policies.
+     * @return map of policies
+     */
+    public Set<PropertyRegenerationPolicy> getPropertyRegenerationPolicies() {
+        if(this.propertyRegenerationPolicies == null) {
+            this.propertyRegenerationPolicies = new HashSet<>();
+            Map<String, Policy> policies = super.getPolicies();
+            if(policies != null) {
+                for (Map.Entry<String, Policy> entry : policies.entrySet()) {
+                    Policy value = entry.getValue();
+                    if(value instanceof PropertyRegenerationPolicy) {
+                        this.propertyRegenerationPolicies.add((PropertyRegenerationPolicy) value);
+                    }
+                }
+            }
+        }
+
+        return this.propertyRegenerationPolicies;
+    }
+
+    /**
+     * Method that allows subclasses to override the type reference with a
+     * subclass.
+     * 
+     * @return
+     */
+    @SuppressWarnings("rawtypes")
+    @Override
+    public Class getDeserializationClass() {
+        return PropertyRegenerationPolicyInput.class;
+    }
+
+    /**
+     * Method that allows subclasses to use an extended type for the policy
+     * class.
+     * 
+     * @param identifier
+     * @return
+     */
+    @Override
+    public DefaultPolicy createPolicy(String policyIdentifier) {
+        return new DefaultPropertyRegenerationPolicy(policyIdentifier);
+    }
+
+    /**
+     * Method that allows subclasses to set any additional configurations while
+     * reading a policy.
+     */
+    @Override
+    public void setAdditionalConfigurations(Policy policy, PolicyInput input) {
```

Comment:
A: might be worth calling out in the docstring params so laymen can know the difference between policy and input

Reply:
A: suggesting to refactor the method name to `addPolicy` unless the current naming still makes more sense

Reply:
This is an override from the abstract class, so can't change the method name